draft-irtf-cfrg-opaque.md

```diff
@@ -428,39 +429,52 @@ with the following API:
   `params` to strengthen the input `msg` against offline dictionary attacks.
   This function also needs to satisfy collision resistance.
 
-## Authenticated Key Exchange (AKE) Protocol {#deps-ake}
+## Key Recovery Method {#deps-keyrec}
 
-OPAQUE additionally depends on an Authenticated Key Exchange (AKE) protocol.
-This specification defines one particular AKE based on 3DH; see {{ake-protocol}}.
-3DH assumes a prime-order group as described in {{I-D.irtf-cfrg-voprf, Section 2.1}}.
-We let `Npk` and `Nsk` denote the size of public and private keys, respectively,
-used in the AKE. The AKE protocol must provide the following functions:
+OPAQUE relies on a key recovery mechanism so that a client can engage in the AKE.
+The key recovery mechanism MUST enable deterministic recovery of authentication material
+compatible with the AKE given the OPRF output and additional data.
+This additional data can be encapsulated in a specific `Envelope` structure, saved on
+the server during registration, and provided to the client during login.
 
-- RecoverPublicKey(private_key): Recover the public key related to the input `private_key`.
-- GenerateAuthKeyPair(): Return a randomly generated private and public key pair. This can be
-  implemented by generating a random private key `sk`, then computing `pk = RecoverPublicKey(sk)`.
-- DeriveAuthKeyPair(seed): Derive a private and public authentication key pair
-  deterministically from the input `seed`.
+The key recovery mechanism must provide the two functions allowing for the construction
+of that `Envelope` structure and key recovery from that same `Envelope` structure.
 
```

Comment:
```suggestion
```

Reply:
typo?

Reply:
It's no longer relevant with the new changes.